c3d-apply/workflow/scripts/seg_to_fcsv.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import nibabel as nib
+import numpy as np
+
+
+def compute_centroid(seg, affine, fid):
+    # Get slice indices matching fiducial
+    x, y, z = np.where(seg == fid)
+
+    # Compute centroid
+    # Better performance than np.mean()
+    centroid = [np.sum(x) / len(x), np.sum(y) / len(y), np.sum(z) / len(z)]
+
+    # Return centroid in world coordinates
+    return affine[:3, :3].dot(centroid) + affine[:3, 3]
+
+
+### MAIN ###
+# Grab data and affine
+prob_seg = nib.load(snakemake.input.seg)
+prob_seg_affine = prob_seg.affine
+prob_seg_data = prob_seg.get_fdata()
+
+# Read in fcsv template
+with open(snakemake.input.fcsv_template, "r") as f:
+    fcsv = [line.strip() for line in f]
+
+# Loop over fiducials
+for fid in range(1, 33):
+    # Compute centroid in world coordinates
+    centroid_mm = compute_centroid(prob_seg_data, prob_seg_affine, fid)
+
+    # Update fcsv, skipping header lines
+    fcsv[2 + fid] = fcsv[2 + fid].replace(f"afid{fid}_x", str(centroid_mm[0]))
```

Comment:
Nitpick: My preference would be to define something like `idx_line = 2 + fid` and use that instead of `2 + fid` everywhere.

Reply:
Agree with this too. I had just copy and pasted what I had working from a jupyter notebook and hadn't gone through to make edits.

Reply:
Fixed in [1aa6edb](https://github.com/afids/afids-auto/pull/4/commits/1aa6edbb6dfa215243d95cb9daee7247122df768)